awx/main/tasks.py

```diff
@@ -1730,16 +1730,33 @@ def pre_run_hook(self, instance, **kwargs):
     def post_run_hook(self, instance, status, **kwargs):
         self.release_lock(instance)
         p = instance.project
-        if instance.job_type == 'check' and status not in ('failed', 'canceled',):
+        if instance.should_update_project() and status not in ('failed', 'canceled',):
             fd = open(self.revision_path, 'r')
             lines = fd.readlines()
+            scm_revision = None
             if lines:
-                p.scm_revision = lines[0].strip()
+                scm_revision = lines[0].strip()
             else:
                 logger.info(six.text_type("{} Could not find scm revision in check").format(instance.log_format))
-            p.playbook_files = p.playbooks
-            p.inventory_files = p.inventories
-            p.save()
+            playbook_files = p.playbooks
+            inventory_files = p.inventories
+            if p.status == 'never updated':
+                # Action happens without direction of task manager, so carefully
+                # save in a non-conflicting way with other updates
+                with transaction.atomic():
+                    p = Project.objects.select_for_update().get(id=p.pk)
```

Comment:
@chrismeyersfsu this is where things got a little weird. This code is a consequence of me trying to think through the scenarios. In this case, we are really looking to update the revision with an update that doesn't hold the task manager lock. However, this only applies for a first-ever update, so there's a strange balance of concerns here.